### DISCUSSION TITLE
Remove error from row.getValue when column is not found or create new function

### DISCUSSION THREAD

**Discussion by TommysG** [TanStack/table#5505] [General]
Hello. In my custom table i have some quick filters in the table's toolbar. To show them i have something like this:

```ts
table.getColumn('status') && <TableStatusFilter/>
```

The problem in this is that in my console, i receiver this error when column doesn't exist.
" [Table] Column with id 'status' does not exist."

How about removing the error  or just creating a new function to check if column exists without showing the error. It's very irritating even if a dev mode seeing all these errors.

**Comment by Maraco91** [comment]
Hello Tommy,

try like this

```typescript
table.getAllColumns().find(x => x.id === 'status') && <TableStatusFilter/>
```

**Reply by mikekobit-stripe** [reply]
I guess you can do this? Why have an [API that let's you get a column by ID](https://tanstack.com/table/v8/docs/api/core/table#getcolumn) and by its definition returns `Column<TData> | undefined` but has a side effect of logging an error?

It is quite odd from an API perspective to return an expected value and then side effect of logging an error as if something was broken.

**Comment by mikekobit-stripe** [comment]
I opened https://github.com/TanStack/table/pull/5964 in the hopes of removing the error log, since the type return indicates that the column returned can be `undefined`.

**Reply by mikekobit-stripe** [reply]
While that pull request was opened over a year ago, it was rejected yesterday. [The comment states:](https://github.com/TanStack/table/pull/5964#issuecomment-5181390998)

> You shouldn't have code that can try to call invalid column ids. We even used to throw an error here

Standard JS/TS lookups, such as [`Map.prototype.get()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get), [`document.getElementById()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/getElementById), or [`URLSearchParams.get()`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get), all return `undefined` or `null` silently when a key isn't found.

Having a getter return `undefined` in its type signature but trigger a dev warning at runtime makes the API behave differently than standard JS lookups. The caller already has to check for `undefined`, so logging a warning forces extra pre-validation work just to avoid console spam. I think this is quite frankly the wrong tradeoff between supposed safety and standard API expectations.
